src/device.ts

```diff
@@ -50,3 +50,10 @@ export const getDiskSpaceInfo = (): Promise<{
     postMessageToNativeApp({
         type: 'GET_DISK_SPACE_INFO',
     });
+
+export const getEsimInfo = (): Promise<{
+    supportsEsim: boolean;
+}> =>
+    postMessageToNativeApp({
```

Comment:
What do we want this to return if the native app doesn't support the bridge method?
With this implementation, it will fail the promise. Don't we prefer to return false in that case?
```js
postMessageToNativeApp({
  type: 'GET_ESIM_INFO'
}).catch(() => ({
  supportsEsim: false,
}))
```

Reply:
makes sense, yes.

Changing it